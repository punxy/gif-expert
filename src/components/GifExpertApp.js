import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)
   
    // const handleAdd = () => {
    //     //setCategories([ ...categories, 'HunterXHunter' ]);
    //     setCategories( cats =>  [...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr />

            <AddCategory setCategories={setCategories} />

            <ol>
                { categories.map( category => 
                        <GifGrid key={category} category={category} />
                )}
            </ol>
        </>
    )
}
