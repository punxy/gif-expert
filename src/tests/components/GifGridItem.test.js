import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Test in GifGridItem components', () => {

    const title = 'este es un titulo';
    const url = 'https://localhost/someimg.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('should be render a component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have a image with de url and title from props', () => {
        const img = wrapper.find('img');
        const props = img.props();

        expect(props.src).toBe(url);
        expect(props.alt).toBe(title);
    });

    test('should have a class "animate__fadeIn"', () => {
        const div = wrapper.find('div');
        expect( div.hasClass('animate__fadeIn') ).toBeTruthy();
    });

})