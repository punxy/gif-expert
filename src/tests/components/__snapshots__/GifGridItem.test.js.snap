// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGridItem components should be render a component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="este es un titulo"
    src="https://localhost/someimg.jpg"
  />
  <p>
    este es un titulo
  </p>
</div>
`;
